stan/math/prim/fun/identity_constrain.hpp

```diff
@@ -13,32 +13,33 @@ namespace math {
  * <p>This method is effectively a no-op and is mainly useful as a
  * placeholder in auto-generated code.
  *
- * @tparam T Type of scalar.
+ * @tparam Scalar Type of scalar.
  * @param[in] x free scalar
  * @return transformed input
  */
-template <typename T>
-inline T identity_constrain(const T& x) {
-  return x;
+template <typename Scalar, require_all_stan_scalar_t<Scalar>* = nullptr>
+inline decltype(auto) identity_constrain(Scalar&& x) {
```

Comment:
I'm going to make/run benchmarks on passing by value and reference for stan autodiff types tmrw. My gut here is that since passing var by value requires adding something to the autodiff stack (I think that also generates another `chain()` call) passing by reference may be better

Reply:
Any results from benchmarking?